pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -228,6 +228,23 @@ stdenv.mkDerivation ({
         configureFlags+=" --extra-lib-dirs=$p/lib"
       fi
     done
+  '' + (optionalString stdenv.isDarwin ''
```

Comment:
It's a no-op in that case since static libraries have the different extension `*.a` and aren't affected by the `dynamic-library-dirs` field.